annotation/consume_trigger.go

```diff
@@ -71,9 +77,63 @@ type Prestring interface {
 	String() string
 }
 
+// Assignment is a struct that represents an assignment to an expression
+type Assignment struct {
+	LHSExprStr string
+	RHSExprStr string
+	Position   token.Position
+}
```

Comment:
What you are suggesting logically makes sense, but there were a couple of reasons why I did not opt for this design choice.
1) `backpropAcrossOneToOneAssignment` which populates the `Assignment` struct object, does not have `*ast.AssignStmt`, but instead only has LHS/RHS `ast.Expr`s.
2) We can of course store `ast.Expr`s here. But since we want to convert the `ast.Expr` to string for printing, we'll additionally need to also store `*analysis.Pass` to facilitate that.

Hence, I thought exposing and storing only the bare minimum objects here might be more desirable. Let me know if you think otherwise. :)

Reply:
I see!

The only concern I have is the memory consumption and artifact size increases (we are already getting issue reports stating that NilAway consumes memory a lot). Let's 

(1) run `make build && bin/nilaway -memprofile ./mem.prof std`, then `go tool pprof -alloc_space ./mem.pprof` and then type `top`, which should show the sum of allocated memory. This should be a good proxy to see the memory consumption. We could simply compare before & after this PR to see if it has a huge impact on memory.

(2) we should run performance validations internally to see if it has an impact on artifact size.

Other than that, this makes sense to me 👍

Reply:
Fair point! I performed both the suggestions above: memory profiling and internal performance evaluation. There was a slight increase in memory use, but well within the acceptable bounds. The performance evaluation did not show any noticeable change. In summary, I have confirmed that the changes in this PR do not incur any significant overhead.